benches/server.rs

```diff
@@ -0,0 +1,99 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+use std::{sync::Arc, time::Instant};
+use wasmtime::*;
+use wasmtime_wasi::{sync::WasiCtxBuilder, WasiCtx};
+
+// Tell rustfmt to skip this module reference; otherwise it can't seem to find it (`cargo fmt` says
+// ".../wasmtime/benches does not exist".
+#[rustfmt::skip]
+mod common;
+
+struct Server {
+    permits: tokio::sync::Semaphore,
+    engine: Engine,
+    modules: Vec<Module>,
+    instance_pres: Vec<InstancePre<WasiCtx>>,
+}
+
+impl Server {
+    async fn job(self: Arc<Self>, index: usize) {
+        let _permit = self.permits.acquire().await.unwrap();
+        let ipre = &self.instance_pres[index % self.modules.len()];
+        let wasi = WasiCtxBuilder::new().build();
+        let mut store = Store::new(&self.engine, wasi);
+        let instance = ipre.instantiate_async(&mut store).await.unwrap();
+        let start_func = instance.get_func(&mut store, "_start").unwrap();
```

Comment:
For benchmarking I think it might be better to use `get_typed_func` here which is more expected in server-like situations.